### PR TITLE
Configure connect group and add examples

### PR DESCRIPTION
**Pull request by turkenf**
### Description of your changes

This PR configures resources in the connect group:

- [`aws_connect_bot_association`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/connect_bot_association)
- [`aws_connect_contact_flow`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/connect_contact_flow)
- [`aws_connect_contact_flow_module`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/connect_contact_flow_module)
- [`aws_connect_hours_of_operation`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/connect_hours_of_operation)
- [`aws_connect_instance`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/connect_instance)
- [`aws_connect_lambda_function_association`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/connect_lambda_function_association)
- [`aws_connect_queue`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/connect_queue)
- [`aws_connect_quick_connect`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/connect_quick_connect)
- [`aws_connect_routing_profile`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/connect_routing_profile)
- [`aws_connect_security_profile`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/connect_security_profile)
- [`aws_connect_user_hierarchy_structure`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/connect_user_hierarchy_structure)

Fixes: https://github.com/upbound/provider-aws/issues/67

Note: I couldn't configure `aws_connect_user_hierarchy_group` I reported an issue for this: https://github.com/upbound/provider-aws/issues/120

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

`make uptest-local EXAMPLE_LIST=examples/connect/botassociation.yaml`
`make uptest-local EXAMPLE_LIST=examples/connect/contactflow.yaml`
`make uptest-local EXAMPLE_LIST=examples/connect/contactflowmodule.yaml`
`make uptest-local EXAMPLE_LIST=examples/connect/hoursofoperation.yaml`
`make uptest-local EXAMPLE_LIST=examples/connect/instance.yaml`
`make uptest-local EXAMPLE_LIST=examples/connect/lambdafunctionassociation.yaml`
`make uptest-local EXAMPLE_LIST=examples/connect/queue.yaml`
`make uptest-local EXAMPLE_LIST=examples/connect/quickconnect.yaml`
`make uptest-local EXAMPLE_LIST=examples/connect/routingprofile.yaml`
`make uptest-local EXAMPLE_LIST=examples/connect/securityprofile.yaml`
`make uptest-local EXAMPLE_LIST=examples/connect/userhierarchystructure.yaml`

### CODE REVIEW

**Comment by turkenf** [comment]
/test-examples="examples/connect/botassociation.yaml,examples/connect/contactflow.yaml,examples/connect/contactflowmodule.yaml,examples/connect/hoursofoperation.yaml,examples/connect/instance.yaml,examples/connect/lambdafunctionassociation.yaml,examples/connect/queue.yaml,examples/connect/quickconnect.yaml,examples/connect/routingprofile.yaml,examples/connect/securityprofile.yaml,examples/connect/userhierarchystructure.yaml"

**Comment by turkenh** [comment]
/test-examples="examples/connect/botassociation.yaml,examples/connect/contactflow.yaml,examples/connect/contactflowmodule.yaml,examples/connect/hoursofoperation.yaml,examples/connect/instance.yaml,examples/connect/lambdafunctionassociation.yaml,examples/connect/queue.yaml,examples/connect/quickconnect.yaml,examples/connect/routingprofile.yaml,examples/connect/securityprofile.yaml,examples/connect/userhierarchystructure.yaml"

File: examples/connect/lambdafunctionassociation.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    s3Bucket: upbound-provider-test-data
+    s3Key: hello-python.zip
+    handler: index.py
+    roleRef:
+      name: sample-role
+    runtime: python3.9
+---
+apiVersion: iam.aws.upbound.io/v1beta1
+kind: Role
+metadata:
```
Comment (ezgidemirel) [inline examples/connect/lambdafunctionassociation.yaml:42]: could you please add the following annotation:
```
annotations:
    meta.upbound.io/example-id: connect/v1beta1/lambdafunctionassociation
```

File: examples/lambda/function.yaml [hunk 2]
```diff
@@ -1,16 +1,43 @@
...
+    handler: index.py
     roleRef:
-      name: iam-for-lambda
-    runtime: nodejs12.x
+      name: sample-role
+    runtime: python3.9
+---
+apiVersion: iam.aws.upbound.io/v1beta1
+kind: Role
+metadata:
```
Comment (ezgidemirel) [inline examples/lambda/function.yaml:24]: could you please add the following annotation: 
```
annotations:
    meta.upbound.io/example-id: connect/v1beta1/function
```

**Review by ezgidemirel — COMMENTED** [review]
Thanks @turkenf for your PR! Added two small comments for adding `meta.upbound.io/example-id` annotation.

File: examples/connect/lambdafunctionassociation.yaml (hunk 1 above)
Comment (turkenf) [reply examples/connect/lambdafunctionassociation.yaml:42]: I have added the specified annotation.

File: examples/lambda/function.yaml (hunk 2 above)
Comment (turkenf) [reply examples/lambda/function.yaml:24]: I have added the specified annotation.

**Review by ezgidemirel — APPROVED** [review]
Thanks @turkenf!

**Pull request merged**